src/Mvc/Mvc.Razor/src/TagHelpers/UrlResolutionTagHelper.cs

```diff
@@ -322,9 +320,20 @@ private static string CreateTrimmedString(string input, int start)
         return input.Substring(start, len);
     }
 
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
     private static bool IsCharWhitespace(char ch)
     {
-        return ValidAttributeWhitespaceChars.AsSpan().IndexOf(ch) != -1;
+        if (ch > 32) return false;
+
+        // Valid whitespace characters defined by the HTML5 spec.
+        const long BitMask =
```

Comment:
The implementation here and the one below are quite similar...I'd like to move them to a common helper, but didn't find a place which is appropriate -- please give an advice.